Fix rendering for `render_mode=human`

qgallouedec opened the pull request:
Close #54

qgallouedec commented:
>So maybe we should not enable RGB-rendering with render_mode='human' but instead enable (optionally) high-quality OpenGL rendering with render_mode='rgb_array'?

something like that?

```python
gym.make("PandaReach-v3", render_mode = "rgb_array", "renderer"="OpenGL")
```

The ideal would be to find a way to run OpenGL in headless mode.

jonasreiher commented:
> something like that?
> ```
> gym.make("PandaReach-v3", render_mode="rgb_array", "renderer"="OpenGL")
> ```

Exactly.

I agree that headless OpenGL would be ideal. I'm currently using pyvirtualdisplay for headless runs but rendering is again about four times slower this way. I have no idea how to accomplish this, though.

qgallouedec commented:
From what I can see, it's far from being easy. Let's add the renderer arg in make in the meanwhile. Default will be "tiny".

qgallouedec commented:
I've moved the render args into the constructor. I think this is the proper way to proceed considering the gymnasium doc.

```python
import gymnasium as gym
import panda_gym

env = gym.make("PandaReach-v3", render_mode="rgb_array", render_width=84, render_height=84)
env.reset()
image = env.render()
assert image.shape == (84, 84, 3)
```

I've also added the renderer arg:


```python
import gymnasium as gym
import panda_gym

env = gym.make("PandaReach-v3", render_mode="rgb_array", renderer="OpenGL")
env.reset()
image = env.render()
assert image.shape == (480, 720, 3)
```

jonasreiher commented on panda_gym/pybullet.py:
```diff
@@ -108,11 +105,10 @@ def render(
         """Render.
 
-        If render mode is "rgb_array", return an RGB array of the scene. Else, do nothing.
+        Return an RGB array of the scene.
```
This needs to be changed back. An array is only returned for render more `rgb_array`.

jonasreiher commented on panda_gym/pybullet.py:
```diff
@@ -129,24 +125,7 @@ def render(
-                "a future version. Use the 'render_mode' argument of the constructor instead.",
-                DeprecationWarning,
-            )
-        target_position = target_position if target_position is not None else np.zeros(3)
```
This line has to stay, setting the target position to [0,0,0], in case anyone calls `env.sim.render()` directly.

jonasreiher commented on panda_gym/envs/panda_tasks.py:
```diff
@@ -22,27 +21,46 @@ class PandaFlipEnv(RobotTaskEnv):
             Defaults to "ee".
-        render (bool, optional): Deprecated: This argument is deprecated and will be removed in a future
-            version. Use the render_mode argument instead.
+        renderer (str, optional): Renderer, either "Tiny" or OpenGL". Default to "Tiny".
```
Missing s: Default**s**

Also, maybe mention that only OpenGL is available for `render_mode='human'`.

Applies to all tasks here.

jonasreiher commented on panda_gym/pybullet.py:
```diff
@@ -19,24 +18,17 @@ class PyBullet:
             Defaults to np.array([223, 54, 45]).
-        render (bool, optional): Deprecated: This argument is deprecated and will be removed in a future
-            version. Use the render_mode argument instead.
+        renderer (str, optional): Renderer, either "Tiny" or OpenGL". Default to "Tiny".
```
Maybe mention relation to `render_mode` here as well. For 'human', render_mode is ignored/defaults to OpenGL.

jonasreiher reviewed (commented):
Looks very good and useful to me. I only added some minor comments.

qgallouedec commented:
Thanks for your help @jonasreiher. I'm doing some more checking and hope to merge it tomorrow.

jonasreiher commented:
> Thanks for your help @jonasreiher. I'm doing some more checking and hope to merge it tomorrow.

Thank *you*!

qgallouedec commented:
Now it's LGTM :100:

jonasreiher commented:
When is version 3.0.4 with this change expected to be released?

qgallouedec commented:
Right now
